projects/ccf-shared/src/lib/services/data-source/ccf-database.service.ts

```diff
@@ -75,3 +89,58 @@ export abstract class WorkerCCFDatabaseDataSourceService extends CCFDatabaseData
     };
   }
 }
+
+const REMOTE_METHODS: (keyof DataSource)[] = [
+  'getProviderNames',
+  'getDatasetTechnologyNames',
+  'getOntologyTreeModel',
+  'getCellTypeTreeModel',
+  'getBiomarkerTreeModel',
+  'getReferenceOrgans',
+];
+const FILTER_METHODS_ARG_0: (keyof DataSource)[] = [
+  'getTissueBlockResults',
+  'getAggregateResults',
+  'getOntologyTermOccurences',
+  'getCellTypeTermOccurences',
+  'getBiomarkerTermOccurences',
+  'getScene',
+];
+const FILTER_METHODS_ARG_1: (keyof DataSource)[] = [
+  'getReferenceOrganScene',
+];
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MixedCCfDatabaseDatasourceService extends ForwardingDataSource {
+  constructor(
+    private readonly remote: ApiEndpointDataSourceService,
+    private readonly local: CCFDatabaseDataSourceService,
+  ) {
+    super();
+  }
+
+  protected forwardCall<K extends keyof DataSource>(
+    method: K, ...args: Parameters<DataSourceMethod<K>>
+  ): Observable<DataSourceDataType<K>> {
+    type AnyFunction = (...rest: unknown[]) => ObservableInput<unknown>;
+    type Res = Observable<DataSourceDataType<K>>;
+    const source = this.isRemoteCall(method, args) ? this.remote : this.local;
+    return (source[method] as AnyFunction)(...args) as Res;
+  }
+
+  private isRemoteCall(method: keyof DataSource, args: unknown[]): boolean {
+    return (
+      REMOTE_METHODS.includes(method) ||
+      (FILTER_METHODS_ARG_0.includes(method) && this.isDefaultFilter(args[0] as Filter)) ||
```

Comment:
Lets remove the filter method checks since it should always use local for those.